src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngAlgorithmCore.cs

```diff
@@ -66,7 +66,11 @@ public CngKey GetOrGenerateKey(int keySize, CngAlgorithm algorithm)
                     ExportPolicy = CngExportPolicies.AllowPlaintextExport,
                 };
 
-                CngProperty keySizeProperty = new CngProperty(KeyPropertyName.Length, BitConverter.GetBytes(keySize), CngPropertyOptions.None);
+                Span<byte> keySizeBuffer = stackalloc byte[sizeof(int)];
```

Comment:
What about `ReadOnlySpan<byte> keySizeBuffer = MemoryMarshal.AsBytes(MemoryMarshal.CreateReadOnlySpan(ref keySize, 1))`?

Reply:
To be honest, I don't think it's worth bringing in unsafe APIs like `MemoryMarshal`. RSA key generation takes an order of "dozens to hundreds of milliseconds". This is a micro optimization that wouldn't meaningfully move a benchmark.

Removing the allocation is worthwhile though, since that can put pressure on the GC.